crates/common/src/journal.rs

```diff
@@ -93,30 +107,157 @@ impl ProofJournal {
         .concat()
     }
 
-    pub fn decode_packed(encoded: &[u8]) -> Result<Self, anyhow::Error> {
-        Ok(ProofJournal {
-            payout_recipient: encoded[..20].try_into().context("payout_recipient")?,
-            precondition_hash: encoded[20..52].try_into().context("precondition_output")?,
-            l1_head: encoded[52..84].try_into().context("l1_head")?,
-            agreed_l2_output_root: encoded[84..116]
-                .try_into()
-                .context("agreed_l2_output_root")?,
-            claimed_l2_output_root: encoded[116..148]
-                .try_into()
-                .context("claimed_l2_output_root")?,
-            claimed_l2_block_number: u64::from_be_bytes(
-                encoded[148..156]
-                    .try_into()
-                    .context("claimed_l2_block_number")?,
-            ),
-            config_hash: encoded[156..188].try_into().context("config_hash")?,
-            fpvm_image_id: encoded[188..220].try_into().context("fpvm_image_id")?,
-        })
+    /// Decodes a byte slice representing a packed `ProofJournal` structure into its constituent fields.
+    ///
+    /// The method extracts fixed-width byte segments from the provided input slice, interprets them
+    /// as the respective fields of `ProofJournal`, and validates the integrity of the input.
+    ///
+    /// # Arguments
+    ///
+    /// * `encoded` - A byte slice containing the serialized representation of the `ProofJournal` fields in order.
+    ///
+    /// # Returns
+    ///
+    /// * `Ok(Self)` - A successfully decoded `ProofJournal` instance when the input is valid.
+    /// * `Err(anyhow::Error)` - An error indicating a failure to decode or validate the input.
+    ///
+    /// # Expected Encoding Layout
+    ///
+    /// The `encoded` byte slice is expected to follow this layout:
+    ///
+    /// - Bytes `[0..20]`: `payout_recipient` (20 bytes)
+    /// - Bytes `[20..52]`: `precondition_hash` (32 bytes)
+    /// - Bytes `[52..84]`: `l1_head` (32 bytes)
+    /// - Bytes `[84..116]`: `agreed_l2_output_root` (32 bytes)
+    /// - Bytes `[116..148]`: `claimed_l2_output_root` (32 bytes)
+    /// - Bytes `[148..156]`: `claimed_l2_block_number` (8 bytes - `u64` in big-endian format)
+    /// - Bytes `[156..188]`: `config_hash` (32 bytes)
+    /// - Bytes `[188..220]`: `fpvm_image_id` (32 bytes)
+    ///
+    /// # Errors
+    ///
+    /// This method returns errors in the following cases:
+    /// 1. The slice is too short to extract the required fields.
+    /// 2. Conversion operations (`try_into`) on the byte segments fail due to mismatched lengths.
+    /// 3. Context-specific decoding errors such as invalid values in certain fields.
+    ///
+    /// Each error includes a context string describing the field that caused the failure.
+    pub fn decode_packed(encoded: &[u8]) -> Self {
```

Comment:
Unlike the documentation states, this method does **not** return any errors. If there are concrete reasons for wanting this behavior, I suggest having try_decode and decode functions so both options are available. Otherwise, returning an error and giving the user the chance to address it is cleaner.
The documentation needs to be fixed anyways.